vnext/Microsoft.ReactNative.Cxx/TurboModuleProvider.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "TurboModuleProvider.h"
+
+namespace winrt::Microsoft::ReactNative {
+
+// CallInvoker implementation based on JSDispatcher.
+struct AbiCallInvoker final : facebook::react::CallInvoker {
+  AbiCallInvoker(IReactDispatcher const &jsDispatcher) : m_jsDispatcher(jsDispatcher) {}
+
+  void invokeAsync(std::function<void()> &&func) override {
```

Comment:
The internal instance CallInvoker seems to possibly have some different semantics from this. E.g. it seems they flush pending JS work after invocation https://github.com/facebook/react-native/blob/453713187815b023cbc32cd434580a452f3e97ac/ReactCommon/cxxreact/Instance.cpp#L300-L308

Could we add facilities to the ABI boundary to proxy to the internal CallInvoker instead of adding our own impl? Seems like a surefire way to ensure matching semantics.

Reply:
We are already doing it. The JSDispatcherQueue is implemented on top of the CallInvoker. Since the CallInvoker is being deprecated, I do not want to add new ABI API for it. Though, based on the FB implementation it seems that I can remove the invokeSync implementation because it is not supported by their code.